pkg/verifier/verify.go

```diff
@@ -18,85 +19,154 @@ import (
 	"github.com/thanos-io/thanos/pkg/objstore"
 )
 
-type verifierMetrics struct {
-	blocksMarkedForDeletion prometheus.Counter
+type Verifier interface {
+	IssueID() string
+	Verify(ctx context.Context, conf Config, idMatcher func(ulid.ULID) bool) error
+}
+
+type VerifierRepairer interface {
+	IssueID() string
+	VerifyRepair(ctx context.Context, conf Config, idMatcher func(ulid.ULID) bool, repair bool) error
 }
 
-func newVerifierMetrics(reg prometheus.Registerer) *verifierMetrics {
-	var m verifierMetrics
+// Config is an verifier config.
+type Config struct {
+	Logger      log.Logger
```

Comment:
I think this shouldn't be part of the config semantically. We should pass logger and metrics directly, (I know it'd mess up the interface however it's more sound in that way) or rather in `Context` even though it's not safe enough. 

We have similar patterns in the code base, but this is something that disturbs me. We should find a better pattern do pass this type of global dependencies (GlobalContext?).

Reply:
I thought about context but it's also wrong ;p

Reply:
any suggestions?

Reply:
In this particular implementation I'd suggest:
1. Not to log or increment in the implementations of the interfaces. Try to do that on the wrapper modules
2. Or just create another struct and pass that to the implementations. 

WDYT?

Reply:
Or we could use constructors to pass down those dependencies.